File: hackerrank/30-days-of-code/day004.rs
use std::cmp::{max};


impl Person {

  fn new(initial_age: i32) -> Person {

    if initial_age < 0 {
      println!("This person is not valid, setting age to 0.");
    }

    Person { age: max(0, initial_age) }
  }

  fn amIOld(&self) {
    println!("You are {}.",
      match self.age {
        0 ... 12 => "young",
        13 ... 17 => "a teenager",
        _ => "old"
      });
  }

  fn yearPasses(&mut self) {
    self.age += 1;
  }
}

File: hackerrank/30-days-of-code/day004.rb
class Person
  attr_reader :age

  def initialize(initial_age)
    puts "This person is not valid, setting age to 0." if initial_age < 0
    @age = [initial_age, 0].max
  end

  def amIOld
    category = case @age
               when 0 .. 12 then "young"
               when 13 .. 17 then "a teenager"
               else "old"
               end
    puts "You are #{category}."
  end

  def yearPasses
    @age += 1
  end
end
